### ISSUE TITLE
Form fields are deleted on update to Formie 2.0.20

### ISSUE THREAD

**Issue by roland-d** [verbb/formie#1246]
### Describe the bug

We have had form fields deleted happen on 2 different installations where we updated Formie from 2.0.18 to 2.0.20.

### Steps to reproduce

Here is what happened:
1. We are on Formie 2.0.18
2. There a number of forms working just fine
3. We update to Formie 2.0.20
4. Opening a form in the admin gives the error `SQLSTATE[42S22]: Column not found: 1054 Unknown column 'content.field_JobSeniority__c_gytgprkp' in 'field list'`

I have traced it back to the `m221206_000000_fix_syncs.php` migration lines 53 - 73. 

To give you an example, we have a form with these 2 entries in the table:
![image](https://user-images.githubusercontent.com/359377/213451870-00b297c4-9074-4995-975c-74a6ed6ac8dd.png)

In the `fields` table are only these entries with the handle `JobSeniority__c`:
![image](https://user-images.githubusercontent.com/359377/213452025-1fa9f58d-35d1-4446-adca-b996737a1b9e.png)

Since neither of the handles are present in the `fields` table the fields get removed from the forum. On both installations we had the majority of fields removed on all forms with the exception from 1 or 2 test forms.

Luckily we could restore the form tables from backup but I feel uneasy that these fields are removed. Now I wonder if we have some kind of data inconsistency that should not be there.

### Form settings

- Multi-page form: Yes
- Submission Method: Ajax
- Client-side Validation: Yes
- Custom Form Templates: No

### Craft CMS version

4.3.0

### Plugin version

2.0.20

### Multi-site?

No

### Additional context

In general I am wondering why the `fields` table is apparently missing all these fields while the forms work fine pre-update.

In some forms we use the Sync option but not all forms use this.

**Comment by engram-design** [comment]
This seems certainly related to cleaning up synced fields, but clearly something is going wrong with that. 

Any chance of getting a snapshot of your database before the update for testing? Understand if you can't supply it, I'll get a test setup myself. Feel free to send to `support at verbb.io` if you are able to.

**Comment by roland-d** [comment]
@engram-design I have sent you a full database dump of one affected site.

**Comment by engram-design** [comment]
Fortunately able to figure this out, it's as I said related to pruning synced fields which were getting out of control on some installs as they previously weren't being pruning properly when deleted, or when saving a form.

This bug was specific to you, as your fields contained underscores, which isn't the default behaviour for Formie (but you're welcome to do this). The checks we were doing to fetch the field from the content column of a field didn't factor that in.

Fixed for the next release. To get this early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-4 as 2.0.20",
  "...": "..."
}
```

Then run `composer update`.

**Comment by engram-design** [comment]
Fixed in [2.0.21](https://github.com/verbb/formie/releases/tag/2.0.21)
